test/units/modules/test_iptables.py

```diff
@@ -29,17 +29,10 @@ def test_without_required_parameters(mocker):
         side_effect=fail_json,
     )
     set_module_args({})
-    with pytest.raises(AnsibleFailJson) as exc:
+    possible_errors = r'.*(Failed to find required executable|parameter must be specified).*'
```

Comment:
This test only exists because there is custom argument validation logic in the iptables module. That logic isn't needed. Instead, `required_if` can be used with `['flush', False, ['chain']],` to perform the same validation.

With that change, this unit test can simply be removed, as we don't need to have tests for each module to verify the argument validator is working.

Reply:
no, i added the flush as the custom validation the test was dealing with was assuming iptables was not present, but i can change the test another way